temporal-sdk/src/test/java/io/temporal/workflow/AbandonOnCancelActivityTest.java

```diff
@@ -30,42 +30,45 @@
 import io.temporal.client.WorkflowFailedException;
 import io.temporal.client.WorkflowStub;
 import io.temporal.failure.CanceledFailure;
-import io.temporal.testing.TestWorkflowRule;
+import io.temporal.testing.SDKTestWorkflowRule;
+import io.temporal.workflow.shared.TestActivities;
+import io.temporal.workflow.shared.TestOptions;
+import io.temporal.workflow.shared.TestWorkflows;
 import java.time.Duration;
 import org.junit.Assert;
 import org.junit.Rule;
 import org.junit.Test;
 
 public class AbandonOnCancelActivityTest {
 
-  public static final String NAMESPACE = "UnitTest";
-  private final WorkflowTest.TestActivitiesImpl activitiesImpl =
-      new WorkflowTest.TestActivitiesImpl(null);
+  private final TestActivities.TestActivitiesImpl activitiesImpl =
+      new TestActivities.TestActivitiesImpl(null);
 
   @Rule
-  public TestWorkflowRule testWorkflowRule =
-      TestWorkflowRule.newBuilder()
-          .setWorkflowTypes(TestAbandonOnCancelActivity.class)
-          .setActivityImplementations(activitiesImpl)
-          .setUseExternalService(Boolean.parseBoolean(System.getenv("USE_DOCKER_SERVICE")))
-          .setTarget(System.getenv("TEMPORAL_SERVICE_ADDRESS"))
-          .build();
+  public SDKTestWorkflowRule testWorkflowRule =
+      (SDKTestWorkflowRule)
```

Comment:
I would try to avoid casts in normal code. In this particular case, I would change SDKTestWorkflowRule.Builder to not extend TestWorkflowRule.Builder but delegate to it.

Reply:
Yes, by delegation I mean SDKTestWorkflowRule keeping TestWorkflowRule as a field and forwarding all method calls. 

BTW in IntelliJ, you can generate all these forwarding methods easily:

1. Command->N
1. Delegate Methods...
1. Select field which methods should be delegated
1. Select methods that should be delegated.